openhcl/underhill_core/src/nvme_manager.rs

```diff
@@ -60,6 +60,243 @@ enum InnerError {
         #[source]
         source: nvme_driver::NamespaceError,
     },
+    #[cfg(test)]
+    #[error("failed to create mock nvme driver")]
+    MockDriverCreationFailed(#[source] anyhow::Error),
+}
+
+/// The object that the [`NvmeManager`] manages. This is an NVMe driver.
+/// Abstracted away to make it easier to test the [`NvmeManager`].
+#[async_trait]
+pub trait BackingNvmeDriver: Inspect + Send + Sync {
+    async fn namespace(
+        &self,
+        nsid: u32,
+    ) -> Result<nvme_driver::Namespace, nvme_driver::NamespaceError>;
+    async fn save(&mut self) -> anyhow::Result<nvme_driver::NvmeDriverSavedState>;
+    async fn shutdown(&mut self);
+    fn update_servicing_flags(&mut self, do_not_restart: bool);
+}
+
+#[derive(Inspect)]
+struct RealNvmeDriver {
```

Comment:
I'm not a big fan of using the term "Real" for the non-testing version. Could we perhaps just drop "Real" and call this "NvmeDriver"?

Reply:
Fair enough :) Thanks for the feedback!